src/sql/src/kafka_util.rs

```diff
@@ -219,20 +219,20 @@ pub async fn create_consumer(
             &BTreeMap::new(),
         )
         .await
-        .map_err(|e| sql_err!("{}", e))?;
+        .map_err(|e| sql_err!("{:#}", e))?;
     let consumer = Arc::new(consumer);
 
     let context = Arc::clone(consumer.context());
     let owned_topic = String::from(topic);
-    // Wait for a metadata request for up to one second. This greatly
+    // Wait for a metadata request for up to two seconds. This greatly
     // increases the probability that we'll see a connection error if
     // e.g. the hostname was mistyped. librdkafka doesn't expose a
     // better API for asking whether a connection succeeded or failed,
     // unfortunately.
     task::spawn_blocking(move || format!("kafka_get_metadata:{topic}"), {
         let consumer = Arc::clone(&consumer);
         move || {
-            let _ = consumer.fetch_metadata(Some(&owned_topic), Duration::from_secs(1));
+            let _ = consumer.fetch_metadata(Some(&owned_topic), Duration::from_secs(2));
```

Comment:
This sketches me out, FWIW. I really hate the nondeterminism. I feel like it makes things _harder_ to debug than if we just 100% for sure created the source, without checking for errors, and told folks to look in the source status history collection to see if it really worked. I doubt that going from 1s to 2s will really move the needle.

That said, it's fine for this PR, but we should consider either deleting this code, or changing the timeout to be infinite. It seems safe to me to require seeing at least one successful metadata request before allowing a Kafka source to be created.

Reply:
I will add a note here!